scripts/test/Muon/phase_table_widget/phase_table_presenter_test.py

```diff
@@ -0,0 +1,203 @@
+        result = self.presenter.create_parameters_for_cal_muon_phase_algorithm()
+
+        self.assertEquals(result, {'BackwardSpectra': [2, 4, 6, 8, 10], 'FirstGoodData': 0.1, 'ForwardSpectra': [1, 3, 5, 7, 9],
+                                   'InputWorkspace': 'input_workspace_name', 'LastGoodData': 15,
+                                   'DetectorTable': 'input_workspace_name_phase_table'})
+
+    def test_correctly_retrieves_workspace_names_associsated_to_current_runs(self):
+        self.view.set_input_combo_box = mock.MagicMock()
+        self.context.getGroupedWorkspaceNames = mock.MagicMock(return_value=['MUSR22222', 'MUSR44444'])
+
+        self.presenter.update_current_run_list()
+
+        self.view.set_input_combo_box.assert_called_once_with(['MUSR22222', 'MUSR44444'])
+
+    def test_correctly_retrieves_names_of_current_groups(self):
+        self.view.set_group_combo_boxes = mock.MagicMock()
+
+        self.presenter.update_current_groups_list()
+
+        self.view.set_group_combo_boxes.assert_called_once_with(['fwd', 'bwd'])
+
+    @mock.patch('Muon.GUI.Common.phase_table_widget.phase_table_presenter.AnalysisDataService')
+    def test_that_phase_table_added_to_ADS_with_correct_name_and_group(self, mock_data_service):
+        self.context.phase_context.options_dict['input_workspace'] = 'MUSR22222_raw_data_period_1'
+        parameters = self.presenter.create_parameters_for_cal_muon_phase_algorithm()
+        mock_phase_table = mock.MagicMock()
+
+        self.presenter.add_phase_table_to_ADS(parameters, mock_phase_table)
+
+        mock_data_service.addOrReplace.assert_called_once_with(parameters['DetectorTable'], mock_phase_table)
+        mock_data_service.addToGroup.assert_called_once_with('MUSR22222', parameters['DetectorTable'])
+
+    @mock.patch('Muon.GUI.Common.phase_table_widget.phase_table_presenter.run_CalMuonDetectorPhases')
+    def test_handle_calculate_phase_table_clicked_behaves_correctly_for_succesful_calculation(self, run_algorith_mock):
+        detector_table_mock = mock.MagicMock()
+        self.view.set_input_combo_box(['MUSR22222_raw_data_period_1'])
+        self.context.phase_context.options_dict['input_workspace'] = 'MUSR22222_raw_data_period_1'
+        self.presenter.update_view_from_model()
+        parameters = self.presenter.create_parameters_for_cal_muon_phase_algorithm()
+        run_algorith_mock.return_value = detector_table_mock
+        self.presenter.add_phase_table_to_ADS = mock.MagicMock()
+
+        self.presenter.handle_calulate_phase_table_clicked()
+        self.wait_for_thread(self.presenter.calculation_thread)
+
+        self.presenter.add_phase_table_to_ADS.assert_called_once_with(parameters, detector_table_mock)
+        self.assertTrue(self.view.isEnabled())
+
+    @mock.patch('Muon.GUI.Common.phase_table_widget.phase_table_presenter.run_CalMuonDetectorPhases')
+    def test_handle_calculate_phase_table_clicked_behaves_correctly_for_error_calculation(self, run_algorith_mock):
```

Comment:
can we change the test name? To me it suggests it is testing the ability to calculate errors not the behaviour if an error occurs.